server/src/main/kotlin/ServerGameMaster.kt

```diff
@@ -0,0 +1,228 @@
+    }
+}
+
+suspend fun getFinalPoints(gameState: ServerGameState) {
+    if (gameState.licytacjaPoints < 360) {
+        when (val aktualnaLicytacja = gameState.receiveMsg(gameState.licytacjaWinner)) {
+            is Message.LicytacjaAktualna -> {
+                gameState.licytacjaPoints = aktualnaLicytacja.punkty
+                broadcastEveryoneElse(gameState, aktualnaLicytacja, gameState.licytacjaWinner)
+            }
+            else -> throw Exception("Invalid message, expected final LicytacjaAktualna from ${gameState.licytacjaWinner}")
+        }
+    }
+}
+
+suspend fun gameloop(gameState: ServerGameState) {
+    var lewStarter = gameState.licytacjaWinner
+    var licytacjaWinnerPoints = 0
+    repeat(8) {
+        val (nextLewStarter, newLicytacjaWinnerPoints) = play1Round(lewStarter, gameState, licytacjaWinnerPoints)
+        lewStarter = nextLewStarter
+        licytacjaWinnerPoints = newLicytacjaWinnerPoints
+    }
+
+    if (licytacjaWinnerPoints < gameState.licytacjaPoints) {
+        gameState.addPoints(gameState.licytacjaWinner, -gameState.licytacjaPoints)
+    }
+    else {
+        gameState.addPoints(gameState.licytacjaWinner, gameState.licytacjaPoints)
+    }
+}
+
+suspend fun play1Round(lewStarter: Place, gameState: ServerGameState, licytacjaWinnerPoints: Int): Pair<Place, Int> {
+    var playing = lewStarter
+    val cards: Array<Card> = Array(3) { Card(CardRank.Nine, CardSuit.Hearts) }// dummy cards
+    var newLicytacjaWinnerPoints = licytacjaWinnerPoints
+    // get cards from players, calculate winner, points, meldunek, itd
+    repeat(3) {
+        when (val playCard = gameState.receiveMsg(playing)) {
+            is Message.PlayACard -> {
+                cards[playing.ordinal] = playCard.card
+                broadcastEveryoneElse(gameState, playCard, playing)
+                if (playing == lewStarter && playCard.meldunek && (playCard.card.rank == CardRank.King || playCard.card.rank == CardRank.Queen)) {
+                    gameState.atut = playCard.card.suit
+                    if (playing != gameState.licytacjaWinner) {
+                        gameState.addPoints(playing, playCard.card.suit.value)
+                    }
+                    else {
+                        newLicytacjaWinnerPoints += playCard.card.suit.value
+                    }
+                }
+            }
+            else -> throw Exception("Invalid message")
+        }
+        playing = playing.next()
+    }
+    val nextLewStarter = compareCards(cards, lewStarter, gameState.atut)
+    if (nextLewStarter != gameState.licytacjaWinner) {
+        gameState.addPoints(nextLewStarter, calculatePoints(cards))
+    }
+    else {
+        newLicytacjaWinnerPoints = licytacjaWinnerPoints + calculatePoints(cards)
+    }
+    return nextLewStarter to newLicytacjaWinnerPoints
+}
+
+private suspend fun broadcastEveryoneElse(gameState: ServerGameState, message: Message, sender: Place?) = coroutineScope {
+    for(player in allPlayerPlaces) {
```

Comment:
```suggestion
    for (player in allPlayerPlaces) {
```
kod niepoformatowany